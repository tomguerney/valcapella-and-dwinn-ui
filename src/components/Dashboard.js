import React from 'react'
import CssBaseline from '@material-ui/core/CssBaseline'
import Container from '@material-ui/core/Container'
import Track from './Track'
import LinerNotes from './LinerNotes'
import {
  coverArt,
  redBackground1,
  redBackground2,
  redBackground3,
  redBackground4,
  redBackground5,
} from '../assets/images'
import { track1, track2, track3, track4, track5 } from '../assets/audio'

class Dashboard extends React.Component {
  state = {
    openTrackId: 'notrack',
  }
  onTrackSelected = (id) => {
    if (id === this.state.openTrackId) {
      this.setState(() => ({ openTrackId: 'notrack' }))
    } else {
      this.setState(() => ({ openTrackId: id }))
    }
  }
  render() {
    return (
      <React.Fragment>
        <CssBaseline />
        <Container maxWidth='sm' style={{ textAlign: 'center' }}>
          <img src={coverArt} style={{ width: '100%', marginTop: 30 }} />
          <Track
            id='track1'
            title='Act 1'
            backgroundImage={redBackground1}
            audioSource={track1}
            onTrackSelected={this.onTrackSelected}
            openTrackId={this.state.openTrackId}
          />
          <Track
            id='track2'
            title='Act 2'
            backgroundImage={redBackground2}
            audioSource={track2}
            onTrackSelected={this.onTrackSelected}
            openTrackId={this.state.openTrackId}
          />
          <Track
            id='track3'
            title='Act 3'
            backgroundImage={redBackground3}
            audioSource={track3}
            onTrackSelected={this.onTrackSelected}
            openTrackId={this.state.openTrackId}
          />
          <Track
            id='track4'
            title='Act 4'
            backgroundImage={redBackground4}
            audioSource={track4}
            onTrackSelected={this.onTrackSelected}
            openTrackId={this.state.openTrackId}
          />
          <Track
            id='track5'
            title='Act 5'
            backgroundImage={redBackground5}
            audioSource={track5}
            onTrackSelected={this.onTrackSelected}
            openTrackId={this.state.openTrackId}
          />
          <LinerNotes />
        </Container>
      </React.Fragment>
    )
  }
}

export default Dashboard
