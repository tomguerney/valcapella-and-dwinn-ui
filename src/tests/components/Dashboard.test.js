import React from 'react'
import { shallow } from 'enzyme'
import Dashboard from '../../components/Dashboard'

let wrapper, instance

beforeEach(() => {
  wrapper = shallow(<Dashboard />)
  instance = wrapper.instance()
})

test('should render Dashboard correctly', () => {
  expect(wrapper).toMatchSnapshot()
})

test('should handle onTrackSelected with new id correctly', () => {
  const selectedId = 'track3'
  instance.onTrackSelected(selectedId)
  expect(wrapper.state('openTrackId')).toBe(selectedId)
})

test('should handle onTrackSelected with same id correctly', () => {
  const openTrackId = 'track4'
  wrapper.setState({ openTrackId })
  instance.onTrackSelected(openTrackId)
  expect(wrapper.state('openTrackId')).toBe('notrack')
})
