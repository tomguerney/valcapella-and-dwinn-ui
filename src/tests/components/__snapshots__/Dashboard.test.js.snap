// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Dashboard correctly 1`] = `
<Fragment>
  <WithStyles(CssBaseline) />
  <WithStyles(ForwardRef(Container))
    maxWidth="sm"
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    <img
      src="test-file-stub"
      style={
        Object {
          "marginTop": 30,
          "width": "100%",
        }
      }
    />
    <Track
      audioSource="test-file-stub"
      backgroundImage="test-file-stub"
      id="track1"
      onTrackSelected={[Function]}
      openTrackId="notrack"
      title="Act 1"
    />
    <Track
      audioSource="test-file-stub"
      backgroundImage="test-file-stub"
      id="track2"
      onTrackSelected={[Function]}
      openTrackId="notrack"
      title="Act 2"
    />
    <Track
      audioSource="test-file-stub"
      backgroundImage="test-file-stub"
      id="track3"
      onTrackSelected={[Function]}
      openTrackId="notrack"
      title="Act 3"
    />
    <Track
      audioSource="test-file-stub"
      backgroundImage="test-file-stub"
      id="track4"
      onTrackSelected={[Function]}
      openTrackId="notrack"
      title="Act 4"
    />
    <Track
      audioSource="test-file-stub"
      backgroundImage="test-file-stub"
      id="track5"
      onTrackSelected={[Function]}
      openTrackId="notrack"
      title="Act 5"
    />
    <LinerNotes />
  </WithStyles(ForwardRef(Container))>
</Fragment>
`;
